src/plugins/data/common/exports/export_csv.tsx

```diff
@@ -22,34 +23,54 @@ interface CSVOptions {
   escapeFormulaValues: boolean;
   formatFactory: FormatFactory;
   raw?: boolean;
-  columnsSorting?: string[];
+  sortedColumns?: string[];
+  columnSorting?: EuiDataGridColumnSortingConfig[];
 }
 
 export function datatableToCSV(
   { columns, rows }: Datatable,
-  { csvSeparator, quoteValues, formatFactory, raw, escapeFormulaValues, columnsSorting }: CSVOptions
+  {
+    csvSeparator,
+    quoteValues,
+    formatFactory,
+    raw,
+    escapeFormulaValues,
+    sortedColumns,
+    columnSorting,
+  }: CSVOptions
 ) {
   const escapeValues = createEscapeValue({
     separator: csvSeparator,
     quoteValues,
     escapeFormulaValues,
   });
 
-  const sortedIds = columnsSorting || columns.map((col) => col.id);
+  const sortedIds = sortedColumns
+    ? columns
+        .map((c) => {
+          // need to find original id for transposed column
+          const sortIndex = sortedColumns.findIndex((id) => c.id.endsWith(id));
+          return {
+            id: c.id,
+            sortIndex,
+            isTransposed: (sortedColumns[sortIndex] ?? '') !== c.id,
+          };
+        })
```

Comment:
This is quadratic in terms of time.
Can you use a lookup table here in order to reduce this from `N^2` to `2 * N`?

I think the transposed logic should not be exposed here, rather the `sortedColumns` should be computed on the Lens side where it is aaware of the transposed state, and able to extract the original id with the proper function for it.

Reply:
Removed column sorting logic from export logic in dfc222168209df1b7acb6d9ebeabc49e7d4def21